parallel_algorithms/P3179/P3179.bs

```diff
@@ -713,14 +727,2242 @@ Here are questions we want to answer (potentially not a complete list):
+           && output_iterator<O, const T2&>
+  constexpr ranges::replace_copy_result<borrowed_iterator_t<R>, O>
+    ranges::replace_copy(R&& r, O result, const T1& old_value, const T2& new_value,
+                         Proj proj = {});
+</xmp>
+
+<ins>
+<xmp class="wordingStyle">
+template<class ExecutionPolicy, random_access_iterator I, sized_sentinel_for<I> S, random_access_iterator O,
+         class Proj = identity, class T1 = projected_value_t<I, Proj>, class T2 = T1>
+  requires indirectly_copyable<I, O> && indirectly_writable<O, const T2&> &&
+           indirect_binary_predicate<ranges::equal_to, projected<I, Proj>, const T1*>
+  ranges::replace_copy_result<I, O>
+    ranges::replace_copy(ExecutionPolicy&& exec, I first, S last, O result,
+                         const T1& old_value, const T2& new_value, Proj proj = {});
+template<class ExecutionPolicy, random_access_range R, random_access_iterator O, class Proj = identity,
+         class T1 = projected_value_t<iterator_t<R>, Proj>, class T2 = T1>
+  requires indirectly_copyable<iterator_t<R>, O> && indirectly_writable<O, const T2&> &&
+           indirect_binary_predicate<ranges::equal_to, projected<iterator_t<R>, Proj>, const T1*> &&
+           sized_range<R>
+  ranges::replace_copy_result<borrowed_iterator_t<R>, O>
+    ranges::replace_copy(ExecutionPolicy&& exec, R&& r, O result,
+                         const T1& old_value, const T2& new_value, Proj proj = {});
+</xmp>
+</ins>
+
+<xmp class="wordingStyle">
+template<input_iterator I, sentinel_for<I> S,class O, class T = iter_value_t<O>,
+         class Proj = identity, indirect_unary_predicate<projected<I, Proj>> Pred>
+  requires indirectly_copyable<I, O> && output_iterator<O, const T&>
+  constexpr ranges::replace_copy_if_result<I, O>
+    ranges::replace_copy_if(I first, S last, O result, Pred pred, const T& new_value,
+                            Proj proj = {});
+template<input_range R, class O, class T = iter_value_t<O>, class Proj = identity,
+         indirect_unary_predicate<projected<iterator_t<R>, Proj>> Pred>
+  requires indirectly_copyable<iterator_t<R>, O> && output_iterator<O, const T&>
+  constexpr ranges::replace_copy_if_result<borrowed_iterator_t<R>, O>
+    ranges::replace_copy_if(R&& r, O result, Pred pred, const T& new_value,
+                            Proj proj = {});
+</xmp>
+!!! TODO: review the constrains carefully !!!
+<ins>
+<xmp class="wordingStyle">
+template<class ExecutionPolicy, random_access_iterator I, sized_sentinel_for<I> S, random_access_iterator O,
+         class Proj = identity,
+         class T = projected_value_t<I, Proj>, indirect_unary_predicate<projected<I, Proj>> Pred>
+  requires indirectly_copyable<I, O> && indirectly_writable<O, const T&>
+  ranges::replace_copy_if_result<I, O>
+    ranges::replace_copy_if(ExecutionPolicy&& exec, I first, S last, O result,
+                            Pred pred, const T& new_value, Proj proj = {});
```

Comment:
You change the default type for `T` from being defined by output (`T = iter_value_t<O>`) to being defined by input (`T = projected_value_t<I, Proj>`). Is it an oversight or a deliberate choice?

Reply:
An oversight, thanks.